In [2]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [3]:
import pandas as pd

In [4]:
df1 = pd.read_csv('/content/drive/MyDrive/Coding Dojo/Raw Data/heart_disease_health_indicators_BRFSS2015.csv')
df1.head()

HeartDiseaseorAttack  HighBP  HighChol  ...   Age  Education  Income
0                   0.0     1.0       1.0  ...   9.0        4.0     3.0
1                   0.0     0.0       0.0  ...   7.0        6.0     1.0
2                   0.0     1.0       1.0  ...   9.0        4.0     8.0
3                   0.0     1.0       0.0  ...  11.0        3.0     6.0
4                   0.0     1.0       1.0  ...  11.0        5.0     4.0

[5 rows x 22 columns]

In [5]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 253680 entries, 0 to 253679
Data columns (total 22 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   HeartDiseaseorAttack  253680 non-null  float64
 1   HighBP                253680 non-null  float64
 2   HighChol              253680 non-null  float64
 3   CholCheck             253680 non-null  float64
 4   BMI                   253680 non-null  float64
 5   Smoker                253680 non-null  float64
 6   Stroke                253680 non-null  float64
 7   Diabetes              253680 non-null  float64
 8   PhysActivity          253680 non-null  float64
 9   Fruits                253680 non-null  float64
 10  Veggies               253680 non-null  float64
 11  HvyAlcoholConsump     253680 non-null  float64
 12  AnyHealthcare         253680 non-null  float64
 13  NoDocbcCost           253680 non-null  float64
 14  GenHlth               253680 non-null  float64
 15  

In [6]:
df1['HeartDiseaseorAttack'].value_counts(normalize = 'true')

0.0    0.905814
1.0    0.094186
Name: HeartDiseaseorAttack, dtype: float64

1) Source of data: https://www.kaggle.com/alexteboul/heart-disease-health-indicators-dataset

2) Brief description of data. Heart Disease is the leading cause of death in the United States. This dataset contains 250000 survey responses from 2015 each containing data related to health-related risk behaviors, chronic health conditions, and the use of preventative services.  There is a strong class imbalance: less than 10% of the sample has experienced the conditions we are trying to predict.

3) What is the target? HeartDiseaseorAttack

4) Is this a classification or regression problem? Classification

5) How many features? 20 features

6) How many rows of data? 253680 rows

7) What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset? This is by far the largest dataset I have worked with to date.  I don't yet know what challenges that brings, but I forsee there being something.  The human heart is also a difficult topic to translate to those who may not already know.  It may be a challenge to accurately describe both the model and the results in terms anyone can understand.

In [7]:
df2 = pd.read_csv('/content/drive/MyDrive/Coding Dojo/Raw Data/diabetes_binary_5050split_health_indicators_BRFSS2015.csv')
df2.head()

Diabetes_binary  HighBP  HighChol  CholCheck  ...  Sex   Age  Education  Income
0              0.0     1.0       0.0        1.0  ...  1.0   4.0        6.0     8.0
1              0.0     1.0       1.0        1.0  ...  1.0  12.0        6.0     8.0
2              0.0     0.0       0.0        1.0  ...  1.0  13.0        6.0     8.0
3              0.0     1.0       1.0        1.0  ...  1.0  11.0        6.0     8.0
4              0.0     0.0       0.0        1.0  ...  0.0   8.0        5.0     8.0

[5 rows x 22 columns]

In [8]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 70692 entries, 0 to 70691
Data columns (total 22 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Diabetes_binary       70692 non-null  float64
 1   HighBP                70692 non-null  float64
 2   HighChol              70692 non-null  float64
 3   CholCheck             70692 non-null  float64
 4   BMI                   70692 non-null  float64
 5   Smoker                70692 non-null  float64
 6   Stroke                70692 non-null  float64
 7   HeartDiseaseorAttack  70692 non-null  float64
 8   PhysActivity          70692 non-null  float64
 9   Fruits                70692 non-null  float64
 10  Veggies               70692 non-null  float64
 11  HvyAlcoholConsump     70692 non-null  float64
 12  AnyHealthcare         70692 non-null  float64
 13  NoDocbcCost           70692 non-null  float64
 14  GenHlth               70692 non-null  float64
 15  MentHlth           

In [9]:
df2['Diabetes_binary'].value_counts()

1.0    35346
0.0    35346
Name: Diabetes_binary, dtype: int64

1) source of data. https://www.kaggle.com/alexteboul/diabetes-health-indicators-dataset?select=diabetes_binary_5050split_health_indicators_BRFSS2015.csv

2) brief description of data.  Diabetes is one of the most common chronic diseases in the United States.  This dataset contains survey results from the Behavioral Risk Factor Surveillance System collected by the CDC in 2015.  The original dataset contains nearly 450000 samples with 330 features.  This balanced dataset contains over 70000 samples with 21 features.  

3) what is the target? Diabetes_binary

4) Is this a classification or regression problem? Classification

5) how many features? 21

6) how many rows of data? 70692

7) what, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset? I anticipate the challenge of this dataset being translating results to everyone.  Diabetes is a complicated topic with many potential influencing factors.  Machine learning is also a potentially complicated topic that can be difficult to translate.